Gems/RecastNavigation/Code/Source/Misc/RecastNavigationMeshComponentController.cpp

```diff
@@ -101,19 +107,24 @@ namespace RecastNavigation
 
         RecastNavigationMeshNotificationBus::Event(m_entityComponentIdPair.GetEntityId(),
             &RecastNavigationMeshNotifications::OnNavigationMeshUpdated, m_entityComponentIdPair.GetEntityId());
+        m_updateInProgress = false;
     }
 
     void RecastNavigationMeshComponentController::UpdateNavigationMeshAsync()
     {
-        AZ_PROFILE_SCOPE(Navigation, "Navigation: UpdateNavigationMeshAsync");
+        bool notInProgress = false;
+        if (m_updateInProgress.compare_exchange_strong(notInProgress, true))
+        {
+            AZ_PROFILE_SCOPE(Navigation, "Navigation: UpdateNavigationMeshAsync");
 
-        RecastNavigationProviderRequestBus::Event(m_entityComponentIdPair.GetEntityId(),
-            &RecastNavigationProviderRequests::CollectGeometryAsync,
-            m_configuration.m_tileSize, aznumeric_cast<float>(m_configuration.m_borderSize) * m_configuration.m_cellSize,
-            [this](AZStd::shared_ptr<TileGeometry> tile)
-            {
-                OnTileProcessedEvent(tile);
-            });
+            RecastNavigationProviderRequestBus::Event(m_entityComponentIdPair.GetEntityId(),
```

Comment:
question: Any chance that there is no provider listening? If not, then it's fine, if yes, then could `m_updateInProgress` always be stuck at true forever?